tensorflow_addons/metrics/f1_scores.py

```diff
@@ -23,124 +23,85 @@
-        self.true_positives = self.add_weight('true_positives', shape=[],
-                                              initializer='zeros',
-                                              dtype=tf.float32)
-        self.false_positives = self.add_weight('false_positives', shape=[],
-                                               initializer='zeros',
-                                               dtype=tf.float32)
-        self.false_negatives = self.add_weight('false_negatives', shape=[],
-                                               initializer='zeros',
-                                               dtype=tf.float32)
-
-    def update_state(self, y_true, y_pred):
-        y_true = tf.cast(y_true, tf.int32)
-        y_pred = tf.cast(y_pred, tf.int32)
-
-        # true positive across column
-        self.true_positives.assign_add(tf.cast(tf.math.count_nonzero(
-            y_pred * y_true, axis=None), tf.float32))
-        # false positive across column
-        self.false_positives.assign_add(tf.cast(tf.math.count_nonzero(
-            y_pred * (y_true - 1), axis=None), tf.float32))
-        # false negative across column
-        self.false_negatives.assign_add(tf.cast(
-            tf.math.count_nonzero((y_pred - 1) * y_true, axis=None),
-            tf.float32))
-
-    def result(self):
-        p_sum = tf.cast(self.true_positives + self.false_positives,
-                        tf.float32)
-        # precision calculation
-        precision_micro = tf.math.divide_no_nan(self.true_positives,
-                                                p_sum)
-
-        r_sum = tf.cast(self.true_positives + self.false_negatives,
-                        tf.float32)
-        # recall calculation
-        recall_micro = tf.math.divide_no_nan(self.true_positives,
-                                             r_sum)
-        # f1 micro score calculation
-        mul_value = 2 * precision_micro * recall_micro
-        add_value = precision_micro + recall_micro
-        f1_micro = tf.math.divide_no_nan(mul_value, add_value)
-        f1_micro = tf.reduce_mean(f1_micro)
-
-        return f1_micro
-
-    def reset_states(self):
-        # reset state of the variables to zero
-        self.true_positives.assign(0)
-        self.false_positives.assign(0)
-        self.false_negatives.assign(0)
-
-
-class F1MacroAndWeighted(Metric):
-    """
-    Calculates F1 macro or weighted based on the user's choice
+    Calculates F1 micro, macro or weighted based on the
+    user's choice
     """
 
     def __init__(self, num_classes, average=None,
-                 name='f1_macro_and_weighted_score', dtype=tf.float32):
-        super(F1MacroAndWeighted, self).__init__(name=name)
+                 name='f1_score', dtype=tf.float32):
+        super(F1Score, self).__init__(name=name)
         self.num_classes = num_classes
-        if average not in (None, 'macro', 'weighted'):
+        if average not in (None, 'micro', 'macro', 'weighted'):
```

Comment:
What happens when `average` is set to `None`?

Reply:
Because it passes the check, but the calculation doesn't consider a case when `average=None`.

Reply:
When the average is set to `None`, F1-score is calculated for each class. This condition is handled in https://github.com/SSaishruthi/addons/blob/9348474f998e3bf0438f541832d98e31ea3fc2db/tensorflow_addons/metrics/f1_scores.py#L173

I got reference for `None` from sklearn. Computed is similar to `Macro` but mean will not be computed.

Reply:
I did not add `None` condition before. Added that today. So we have four conditions now:

Micro/ Macro / weighted /None

Reply:
Also, added a test condition for this.